### PR TITLE
[TDG-23] Added enemy death animation

### PR DESCRIPTION
**Pull request by GianiStatie**

### CODE REVIEW

File: src/units/00_components/AbstractUnit.gd [hunk 1]
```diff
@@ -76,3 +81,7 @@ func look_at_position(target_position):
 		
 func take_damage(value: int):
 	stats.health -= value
+
+func _on_is_dead():
+	health_bar.visible = false
```
Comment (Ratoone) [inline src/units/00_components/AbstractUnit.gd:86]: The stats already has a reference to the health bar, I would put this there

File: src/units/00_components/Stats.gd [hunk 2]
```diff
@@ -20,6 +22,8 @@ func _get_max_health():
 func _set_health(value):
 	health = max(min(value, max_health), 0)
 	health_bar.health = health
+	if health == 0:
+		emit_signal("is_dead")
```
Comment (Ratoone) [inline src/units/00_components/Stats.gd:26]:
```suggestion
	if health == 0:
	        health_bar.visible = false
		emit_signal("is_dead")
```

**Review by Ratoone — COMMENTED** [review]
I just noticed the indentation is 8 spaces, and it makes me cringe 😬

File: src/units/00_components/AbstractUnit.gd (hunk 1 above)
Comment (Ratoone) [reply src/units/00_components/AbstractUnit.gd:86]: On the other hand, I would disable the enemy's hitbox, so the towers don't shoot a dying enemy

Comment (GianiStatie) [reply src/units/00_components/AbstractUnit.gd:86]: stop it! It's already dead!

**Comment by GianiStatie** [comment]
> I just noticed the indentation is 8 spaces, and it makes me cringe 😬

Isn't this only in the code review? In the godot editor it looks ok

**Pull request merged**